src/scss/includes/routeLabel.scss

```diff
@@ -0,0 +1,118 @@
+.routeLabel {
+  background-color: $grey-lighter;
+  color: $grey-semi-darkness;
+  border-radius: 6px;
+  padding: 4px 8px;
+  z-index: 1;
+  cursor: pointer;
+  box-shadow: $shadow;
+  display: flex;
+
+  &-vehicleIcon {
+    display: none;
+    width: 40px;
+    height: 40px;
+    margin-right: 4px;
+    color: white;
+    background-color: $action-blue-base;
+    align-items: center;
+    justify-content: center;
+    font-size: 24px;
+    border-radius: 50%;
+
+    .active & {
+      display: flex;
+    }
+  }
+
+  &-duration {
+    font-size: 18px;
+    font-weight: bold;
+    color: $grey-black;
+
+    .active & {
+      color: $action-blue-base;
+    }
+  }
+
+  &-distance {
+    font-size: 14px;
+  }
+
+  &.active {
+    background-color: white;
+    padding-left: 4px;
+    z-index: 2;
+  }
+
+  &::after {
+    content: '';
+    width: 0;
+    height: 0;
+    position: absolute;
+  }
+
+  &--bottom {
+    top: -6px;
+
+    &::after {
+      border-left: 6px solid transparent;
+      border-right: 6px solid transparent;
+      border-top: 6px solid $grey-lighter;
+      bottom: -6px;
+      left: calc(50% - 6px);
+    }
+
+    &.active::after {
+      border-top-color: white;
+    }
+  }
+
+  &--left {
+    left: 6px;
+
+    &::after {
+      border-top: 6px solid transparent;
+      border-bottom: 6px solid transparent;
+      border-right: 6px solid $grey-lighter;
+      left: -6px;
+      top: calc(50% - 6px);
+    }
+
+    &.active::after {
+      border-right-color: white;
+    }
+  }
+
+  &--top {
+    top: 6px;
+
+    &::after {
+      border-left: 6px solid transparent;
+      border-right: 6px solid transparent;
+      border-bottom: 6px solid $grey-lighter;
+      top: -6px;
+      left: calc(50% - 6px);
+    }
+
+    &.active::after {
+      border-bottom-color: white;
+    }
+  }
+
+  &--right {
+    left: -6px;
+
+    &::after {
```

Comment:
can we factorize these properties using a local `mixin` :lipstick:  ?

Reply:
I tried for some time, I think it's feasible but this is really not easy, as Sass is limited in its management of variables and you can't do it with only one var.
IMO it would be a good improvement for later :)